lib/graphql/schema/field.rb

```diff
@@ -725,37 +725,42 @@ def with_extensions(obj, args, ctx)
         if @extensions.empty?
           yield(obj, args)
         else
-          memos = []
           # This is a hack to get the _last_ value for extended obj and args,
           # in case one of the extensions doesn't `yield`.
           # (There's another implementation that uses multiple-return, but I'm wary of the perf cost of the extra arrays)
-          extended = { args: args, obj: obj }
-          value = run_extensions_before_resolve(memos, obj, args, ctx, extended) do |obj, args|
+          extended = { args: args, obj: obj, memos: nil }
```

Comment:
We can avoid allocating `[]` when no memo values are given (which is the default case)